### PR TITLE
db: move virtual tables into virtual_tables.cc

### PR DESCRIPTION
**Pull request by denesb**
The definitions of virtual tables make up approximately a quarter of the huge system_keyspace.cc file (almost 4K lines), pulling in a lot of headers only used by them.
Move them to a separate source file to make system_keyspace.cc easier for humans and compilers to digest.
This patch also moves the `register_virtual_tables()`, `install_virtual_readers()` as well as the `virtual_tables` global.

### CODE REVIEW

**Comment by denesb** [comment]
> CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/1936/

[Tests / Sanity Tests / batch_test.TestBatch.test_logged_batch_throws_uae](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/1936/testReport/junit/batch_test/TestBatch/Tests___Sanity_Tests___test_logged_batch_throws_uae/) failed with timeout. Re-kicking.

**Comment by denesb** [comment]
@scylladb/scylla-maint ping.

**Review by xemul — APPROVED** [review]
LGTM, but it has merge conflicts, please, rebase

File: db/virtual_tables.cc [hunk 1]
```diff
@@ -0,0 +1,1030 @@
+/*
+ * Modified by ScyllaDB
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
```
Comment (nyh) [inline db/virtual_tables.cc:7]: I think this "Apache-2.0" and "Modified by ScyllaDB" are likely to be a mistake - I don't think anything in this file is from Cassandra, I think the virtual tables was our own invention. But please correct me if I'm wrong.

**Review by kbr-scylla — APPROVED** [review]

File: db/virtual_tables.cc (hunk 1 above)
Comment (denesb) [reply db/virtual_tables.cc:7]: Yes, this is copy-pasta. Will remove.

**Comment by denesb** [comment]
v2:
* Rebased and fixed license blurb.

**Pull request closed without merging**